src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -25,43 +27,63 @@ interface PartyMapProps {
 export const PartyMap: React.FC<PartyMapProps> = ({ venue }) => {
   const { user, profile } = useLiveUser();
 
-  const selfAndPortalSpaceIds = useMemo(() => {
-    const spaceIds = (venue?.rooms ?? [])
-      .map((portal) => portal.spaceId)
-      .filter((spaceId) => !!spaceId) as SpaceId[];
-    return [venue?.id].concat(spaceIds);
-  }, [venue]);
+  const portalRef = useRef<HTMLDivElement | null>(null);
+  const [selectedPortal, setSelectedPortal] = useState<
+    PortalWithBounds | undefined
+  >();
 
   const { events: selfAndChildVenueEvents } = useSpaceEvents({
     worldId: venue.worldId,
-    spaceIds: selfAndPortalSpaceIds,
+    spaceIds: [selectedPortal?.spaceId || ""],
   });
 
-  const [selectedRoom, setSelectedRoom] = useState<Room | undefined>();
-
-  const hasSelectedRoom = !!selectedRoom;
-
-  const selectedRoomEvents = useMemo(() => {
-    if (!selfAndChildVenueEvents || !selectedRoom) return [];
+  const [firstEvent] = useMemo(() => {
+    if (!selfAndChildVenueEvents || !selectedPortal) return [];
 
     return selfAndChildVenueEvents
       .filter(
         (event) =>
-          event.spaceId === selectedRoom.spaceId && isEventLiveOrFuture(event)
+          event.spaceId === selectedPortal.spaceId && isEventLiveOrFuture(event)
       )
-      .sort(eventsByStartUtcSecondsSorter);
-  }, [selfAndChildVenueEvents, selectedRoom]);
-
-  const selectRoom = useCallback((room: Room) => {
-    if (room.type && COVERT_ROOM_TYPES.includes(room.type)) return;
+      .sort(eventTimeAndOrderComparator);
+  }, [selfAndChildVenueEvents, selectedPortal]);
 
-    setSelectedRoom(room);
+  const selectPortal = useCallback((portal: PortalWithBounds) => {
+    setSelectedPortal(portal);
   }, []);
 
-  const unselectRoom = useCallback(() => {
-    setSelectedRoom(undefined);
+  const unselectPortal = useCallback(() => {
+    setSelectedPortal(undefined);
   }, []);
 
+  const isUnclickable =
+    selectedPortal?.visibility === RoomVisibility.unclickable ||
+    selectedPortal?.type === RoomType.unclickable;
+  const isCovertRoom =
+    selectedPortal?.type && COVERT_ROOM_TYPES.includes(selectedPortal?.type);
+  const shouldBeClickable = !isCovertRoom && !isUnclickable;
+  const { enterPortal } = usePortal({ portal: selectedPortal });
+  const analytics = useAnalytics({ venue });
+
+  const [enterWithSound] = useCustomSound(selectedPortal?.enterSound, {
+    interrupt: true,
+    onend: enterPortal,
+  });
+
+  const selectPortalWithSound = useCallback(
+    (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+      if (!shouldBeClickable || !selectedPortal) return;
+      analytics.trackEnterRoomEvent(
+        selectedPortal?.title,
+        selectedPortal?.template
+      );
+      isExternalPortal(selectedPortal)
+        ? openUrl(selectedPortal?.url)
+        : enterWithSound();
```

Comment:
Do we still actually want portal sounds? I don't think they can be configured anywhere.
Also, external portals aren't a thing any more. We can get rid of that logic.

Reply:
thanks, I've removed all external/sound related logic from portal components [b465939](https://github.com/sparkletown/sparkle/pull/3008/commits/b465939a4b776d9b32f1b818b4f4cfa952f7af86)